File: src/main/java/body/core/logger/Loggerable.java
package body.core.logger;

/**
 * Created by vbolinch on 09/02/2017.
 * Trait
 */
public interface Loggerable extends LoggerableParent {

    default void loggerTrace(Object str) { getLogger().trace(str); }

    default void loggerDebug(Object str) { getLogger().debug(str); }

    default void loggerInfo(Object str) {  getLogger().info(str); }

    default void loggerError(Object str) {  getLogger().error(str); }
}

File: src/main/java/body/core/logger/Loggerable.kt
package body.core.logger

/**
 * Created by vbolinch on 09/02/2017.
 * Trait
 */
interface Loggerable : LoggerableParent {

    fun loggerTrace(str: Any)  = get().trace(str)

    fun loggerDebug(str: Any)  = get().debug(str)

    fun loggerInfo(str: Any) = get().info(str)

    fun loggerError(str: Any) = get().error(str)

}


